### PR TITLE
T12098 remove deprecated code

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: code quality
* Link to issue: #12098 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Removed deprecated code based on the list

Thanks

### CODE REVIEW

**Comment by niden** [comment]
cc @sergeyklay

File: CHANGELOG-4.0.md [hunk 1]
```diff
@@ -45,29 +44,51 @@
...
+- Removed deprecated `Phalcon\Mvc\ViewInterface::getParams()`
+- Removed deprecated `Phalcon\Paginator\Adapter\Model::getPaginate()`
+- Removed deprecated `Phalcon\Paginator\Adapter\Model - before`
+- Removed deprecated `Phalcon\Paginator\Adapter\Model - total_pages`
+- Removed deprecated `Phalcon\Paginator\Adapter\NativeArray::getPaginate()`
+- Removed deprecated `Phalcon\Paginator\Adapter\NativeArray - before`
+- Removed deprecated `Phalcon\Paginator\Adapter\NativeArray - total_pages`
+- Removed deprecated `Phalcon\Paginator\Adapter\QueryBuilder::getPaginate()`
+- Removed deprecated `Phalcon\Paginator\Adapter\QueryBuilder - before`
+- Removed deprecated `Phalcon\Paginator\Adapter\QueryBuilder - total_pages`
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: Could you please use common format for vars:
```
Phalcon\Paginator\Adapter\QueryBuilder::$total_pages
```

File: CHANGELOG-4.0.md [hunk 2]
```diff
@@ -45,29 +44,51 @@
...
+- Removed deprecated `Phalcon\Cli\Console::addModules`
+- Removed deprecated `Phalcon\Debug::getMajorVersion`
+- Removed deprecated `Phalcon\Mvc\Model\Criteria::addWhere`
+- Removed deprecated `Phalcon\Mvc\Model\Criteria::order`
 - Removed deprecated `Phalcon\Mvc\Model\Metadata\Apc`
 - Removed deprecated `Phalcon\Mvc\Model\Metadata\Memcache`
 - Removed deprecated `Phalcon\Mvc\Model\Metadata\Xcache`
+- Removed deprecated `Phalcon\Mvc\View::getParams()`
+- Removed deprecated `Phalcon\Mvc\ViewInterface::getParams()`
+- Removed deprecated `Phalcon\Paginator\Adapter\Model::getPaginate()`
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: Could you please use common format for methods:
```
Phalcon\Paginator\Adapter\QueryBuilder::getPaginate
```

File: CHANGELOG-4.0.md [hunk 3]
```diff
@@ -45,29 +44,51 @@
...
-- Removed deprecated `Phalcon\Security::getSslVersionNumber`
-- Removed `Phalcon\Http\RequestInterface::isSoapRequested` in favor of `Phalcon\Http\Request::isSoap`
-- Removed `Phalcon\Http\RequestInterface::isSecureRequest` in favor of `Phalcon\Http\RequestInterface::isSecure`
-- Removed `Phalcon\Validation\MessageInterface` and `Phalcon\Mvc\Model\MessageInterface` in favor of `Phalcon\Messages\MessageInterface`
-- Removed `Phalcon\Validation\Message` and `Phalcon\Mvc\Model\Message` in favor of `Phalcon\Messages\Message`
-- Removed `Phalcon\Validation\Message\Group` in favor of `Phalcon\Messages\Messages`
+- PHP < 7.2 no longer supported
+- Removed `xcache` support from adapters
+- Removed `apc` support from adapters (use `apcu`)
+- Removed `memcache` support from adapters (use `libmemcached`)
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: Is there any issue/PR we can mention here and below for each of change log entry?

File: phalcon/mvc/micro/lazyloader.zep [hunk 4]
```diff
@@ -28,40 +28,37 @@ use Phalcon\Mvc\Model\BinderInterface;
...
 	/**
-	 * Initializes the internal handler, calling functions on it
+	 * Calling __call method
+	 *
+	 * @param  string method
+	 * @param  array arguments
+	 * @return mixed
 	 */
-	public function __call(string! method, array arguments) -> var
+	public function callMethod(string! method, arguments, <BinderInterface> modelBinder = null)
```
Comment (sergeyklay) [inline phalcon/mvc/micro/lazyloader.zep]: Is there issue where we decided to remove `__call`

**Review by sergeyklay — CHANGES_REQUESTED** [review]
Some minor changes

**Review by sergeyklay — APPROVED** [review]

**Pull request merged**